web/static/js/components/surveys/SurveyForm.jsx

```diff
@@ -50,9 +53,18 @@ class SurveyForm extends Component {
   }
 
   launchSurvey() {
-    const { projectId, surveyId, router } = this.props
-    launchSurvey(projectId, surveyId)
-      .then(() => router.push(routes.survey(projectId, surveyId)))
+    const { projectId, survey, router, dispatch } = this.props
+    launchSurvey(projectId, survey.id)
+      .then(() => {
+        if (survey.panelSurveyId) {
+          // An occurrence of the panel survey was launched -> the panel survey has changed.
+          // The Redux store must be updated with the panel survey new state.
+          dispatch(panelSurveyActions.fetchPanelSurvey(survey.projectId, survey.panelSurveyId))
+          dispatch(panelSurveysActions.fetchPanelSurveys(survey.projectId))
```

Comment:
This is the third time in the code we do the exact same thing. I believe is time to extract it 🙂

Reply:
I was expecting this comment 😛 I'll see what can I do 😇